### PR TITLE
Added buttonsOrder option

### LINKED ISSUE #2135: has any effort been made to allow custom ordering of toolbar buttons?
has any effort been made to allow custom ordering of toolbar buttons?

i searched and did not see any. i can make pull request if desired but wanted to make sure that this hasn't been discussed/underway already.

thanks

### PR DESCRIPTION
**Pull request by wenzhixin**
Fix #2135 

Example: https://live.bootstrap-table.com/code/wenzhixin/1191

ref: https://github.com/wenzhixin/bootstrap-table-examples/pull/387

### CODE REVIEW

**Comment by UtechtDustin** [comment]
The buttons will broke, if you move the columns after paginationSwtich in the Foundation theme.
https://live.bootstrap-table.com/code/UtechtDustin/1204

Bildschirmfoto vom 2019-11-19 09-37-22

**Comment by wenzhixin** [comment]
* Fixed foundation dropdown html error.
* Added `data-*` support, for example: 
`data-buttons-order="['columns', 'paginationSwitch', 'refresh', 'toggle', 'fullscreen']"`
`data-buttons-order="[columns, paginationSwitch, refresh, toggle, fullscreen]"`
`data-buttons-order="[columns,paginationSwitch,refresh,toggle,fullscreen]"`

**Comment by wenzhixin** [comment]
image

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -503,78 +503,84 @@ class BootstrapTable {
...
-        </button>`)
-    }
+        </button>`,
 
-    if (o.showFullscreen) {
-      html.push(`<button class="${this.constants.buttonsClass}" type="button" name="fullscreen"
+      fullscreen: `<button class="${this.constants.buttonsClass}" type="button" name="fullscreen"
         aria-label="Fullscreen" title="${o.formatFullscreen()}">
         ${o.showButtonIcons ? Utils.sprintf(this.constants.html.icon, o.iconsPrefix, o.icons.fullscreen) : '' }
         ${o.showButtonText ? o.formatFullscreen() : ''}
```
Comment (djhvscf) [inline src/bootstrap-table.js]: I know this is not part of this PR, but can we change the `o`variable name to something more meaningful like `opts`?

Comment (wenzhixin) [reply src/bootstrap-table.js]: Done!

**Pull request merged**